internal/webhook/inferenceservice/mutating_test.go

```
@@ -0,0 +1,356 @@
+)
+
+const (
+	testNamespace        = "glue-ns"
+	testInferenceService = "glue-isvc"
+	testSecret           = "glue-secret"
+)
+
+type TestCase struct {
+	name               string
+	secretType         string
+	secretData         map[string][]byte
+	secretNamespace    string
+	annotations        map[string]string
+	predictorSpec      map[string]interface{}
+	operation          admissionv1.Operation
+	expectedAllowed    bool
+	expectedMessage    string
+	expectedPatchCheck func([]jsonpatch.JsonPatchOperation) bool
+}
+
+func setupTestEnvironment(t *testing.T) (*runtime.Scheme, context.Context) {
+	t.Helper()
+	sch, err := scheme.New()
+	NewWithT(t).Expect(err).ShouldNot(HaveOccurred())
+
+	return sch, context.Background()
+}
+
+func createWebhook(cli client.Client, sch *runtime.Scheme) *inferenceservice.ConnectionWebhook {
+	webhook := &inferenceservice.ConnectionWebhook{
+		Client:  cli,
+		Decoder: admission.NewDecoder(sch),
+		Name:    "glueisvc-test",
+	}
+	return webhook
+}
+
+func createTestSecret(name, namespace, connectionType string, data map[string][]byte) *corev1.Secret {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				annotations.ConnectionTypeRef: connectionType,
+			},
+		},
+		Data: data,
+	}
+
+	return secret
+}
+
+func createTestInferenceService(name, namespace string, annotations map[string]string, predictorSpec map[string]interface{}) *unstructured.Unstructured {
+	isvc := envtestutil.NewInferenceService(name, namespace)
+	unstructuredISVC, ok := isvc.(*unstructured.Unstructured)
+	if !ok {
+		panic("failed to cast InferenceService to unstructured")
+	}
+
+	if annotations != nil {
+		unstructuredISVC.SetAnnotations(annotations)
+	}
+
+	if len(predictorSpec) > 0 {
+		if err := unstructured.SetNestedMap(unstructuredISVC.Object, predictorSpec, "spec", "predictor"); err != nil {
+			panic("failed to set nested map: " + err.Error())
```

it is a for testing, but probably better to return an error